src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -277,7 +277,7 @@ public ProjectContext Build()
 
                     if (platformDependency != null)
                     {
-                        platformLibrary = libraries[new LibraryKey(platformDependency.Value.Name)];
+                        libraries.TryGetValue(new LibraryKey(platformDependency.Value.Name), out platformLibrary);
```

Comment:
Sure, but assuming the unresolved dependency is not the tool in a project, a ProjectToolsCommandResolver may still try to use the projectcontext returned by this to find the tool. 

What do you think the correct behavior should be there?